capstone-frontend/src/components/ListPage.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import { Button, Spinner } from 'react-bootstrap'
+
+class ListPage extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      gBooks: [],
+      loading: true,
+      empty: false
+    }
+  }
+
+
+  fetchBooks = async () => {
+
+    const bookList = await fetch(`/api/booklist?id=${this.props.match.params.id}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    const gbookIDs = bookList[0].gbookIDs;
```

Comment:
Could `bookList` be empty?

Reply:
No couldn't be empty. The user cannot reach the ListPage if a BookList doesn't exist. On the other hand, a BookList may not contain any books, thus I made a check for that.